### PR TITLE
Create Blog “get-started-with-application-performance-monitoring-tools-overview”

### PR DESCRIPTION
**Pull request by GuopingJia**
Automatically generated by Netlify CMS

### CODE REVIEW

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
+---
+title: "Get started with application performance monitoring: tools overview"
+date: 2022-12-27T08:53:23.176Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+disable: false
+tags:
```
Comment (choukden) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:7]: You may want to add the tag:
hpe-greenlake-for-private-cloud-enterprise

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+* enrich raw data using a variety of analyzers, tokenizer, filters, and enrichment options;
+* support document storage, time series analysis and metrics, and geospatial analytics;
+* provide full-text search capabilities with its inverted index, tunable relevance scoring and advanced query _DSL_;
+* find data relationships through aggregations and graph exploration and create alerts;
+* model and automate the analysis of time series data, combine alerting and inference using machine learning;
+
+Elastic builds and maintains the *Elastic Stack*, an _all-in-one_ platform built upon the proven *ELK Stack* for the logs, metrics, and application trace data with a multitude of out-of-the-box integrations. Elastic Stack is the foundation for its primary solutions, *Elastic Enterprise Search*, the fleet of search solutions, *Elastic Observability*, the solution for unified visibility across logs, metrics and APM data, and *Elastic Security*, the solution that unifies endpoint protection and _SIEM_. You can easily deploy any of these solutions as a managed service with Elastic Cloud, with one stack powering three solutions.
```
Comment (choukden) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:162]: ...upon the proven *Elasticsearch, Logstash, and Kibana (ELK) Stack* for the logs, ...

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+* provide dependency analysis of service instances and endpoins; 
+* support service topology map analysis;
+* detect slow services and endpoints and provide performance optimization;
+* detect slow _SQL_ statement for database performance monitoring;
+* provide message queue performance and consuming latency monitoring;
+* start tracing from browser for browser performance monitoring;
+* support infrastructure monitoring for Kubernetes and Linux;
+* support alerting using rules in both observability analysis language and metric analysis language;
+
+Apache SkyWalking provides a list of agents to be used for building *Java*, *.NET Core*, *PHP*, *Node.js*, *Golang*, *LUA*, *Rust* and *C++* apps. It supports to integrate and collect data from multiple sources, including _Prometheu_, _OpenTelemetry_ and _Zabbix_ for metrics and logs, _Zipkin_ for traces. It provides tracing, metrics analysis, alerting, service mesh observability and visualization.
```
Comment (choukden) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:210]: ...including _Prometheus_, ....

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Guoping, 
Just a couple of suggestions for edits. 

Thank you so much for such great blog post about the common APM tooling.

Dale will go through the editorial review in the coming day or so. Please stay tuned for further suggestions for edits and comments.

Thanks,

Denis

**Comment by GuopingJia** [comment]
> Hi Guoping, Just a couple of suggestions for edits.
> 
> Thank you so much for such great blog post about the common APM tooling.
> 
> Dale will go through the editorial review in the coming day or so. Please stay tuned for further suggestions for edits and comments.
> 
> Could you please confirm this post is the first of a 3 blog post series about APM tooling, and using Skywalking to monitor distributed applications and microservices deployed on K8s clusters?
> 
> Shall I assume you will submit another blog post contribution in the coming weeks to complete this 3 blog post series?
> 
> Thanks,
> 
> Denis

Thanks Denis for your review and comments. 

The new patch has been submitted that includes all your comments. 

Yes, there will be a 3 blog post series about APM tooling and using Apache SkyWalking. 2 blogs have been submitted for your review, #1634 (this blog) and #1636 (for monitoring applications deployed on K8s cluster using agent instrumentation). I've been working on the last one, using SkyWalking for K8s monitoring in HPE GreenLake for Private Cloud Enterprise. This monitoring is using different approach, kube-state-metrics for K8s metrics, OpenTelemetry collector / receiver for K8s monitoring. With another seperate blog from Thirukkannan (for infrastructure monitoring), we will have the full coverage using SkyWalking for HPE GreenLake monitoring.

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+Apache SkyWalking is lightweight, scalable, and supports alerting and visualization. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and removes the restricted security restriction in user environment. 
+
+## Conclusion
+
+This blog post gave a closer look at some of the best APM tools, either open-source or commercial vendor based. It listed the key features of each APM tool, and discussed in details their strengths and weaknesses. The importance of a good APM solution is now indisputable. All it takes is to pick the right one based on your monitoring needs for your applications.   
+
+This blog post is the first of three in a series. In the following posts, I will show you the detailed process how to set up the *Apache SkyWalking*, as the APM tool, in HPE GreenLake for Private Cloud Enterprise for monitoring and alerting both customer application and Kubernetes cluster.
```
Comment (choukden) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:223]: Hi Guoping, now that I better understand what are the blog posts that are part of the series, how about this suggestion for edits:

"This blog post is the first of three in a series. In the following post, I will show you the detailed process on [how to set up the ***Apache SkyWalking*** APM tool in HPE GreenLake for Private Cloud Enterprise](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud/) for monitoring and alerting both customer application and Kubernetes cluster. Finally, I will expand on using Apache SkyWalking APM tool [to monitor the ***Online Boutique*** microservices application deployed across the hybrid cloud environment](https://developer.hpe.com/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh/).

Notice that Dale might have a better idea.

What do you think?
Denis

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Guoping, 
I have just added one additional comment for the conclusion of the post, now that I better understand what are the posts that are part of your APM monitoring series.

Denis

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md (hunk 4 above)
Comment (choukden) [reply content/blog/get-started-with-application-performance-monitoring-tools-overview.md:223]: @GuopingJia 

Based on our discussion in the call today, I think one suggestion would be:

This blog post is the first of three in a series. [In the second post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud/), I will show you the detailed process on how to set up the Apache SkyWalking APM tool for monitoring and alerting of customer applications deployed on Kubernetes cluster provisioned on HPE GreenLake for Private Cloud Enterprise. [In the last (or third) post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will expand on setting up Apache SkyWalking APM tool to monitor the infrastructure of Kubernetes clusters deployed on HPE GreenLake for Private Cloud Enterprise.

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Guoping,

Now that I have a better idea of the two blog post series, I am adding a suggestion for the last paragraph of the Conclusion section.

This blog post is the first of three in a series. [In the second post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud/), I will show you the detailed process on how to set up the Apache SkyWalking APM tool for monitoring and alerting of customer applications deployed on Kubernetes cluster provisioned on HPE GreenLake for Private Cloud Enterprise. [In the last (or third) post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will expand on setting up Apache SkyWalking APM tool to monitor the infrastructure of Kubernetes clusters deployed on HPE GreenLake for Private Cloud Enterprise.

Feel free to adjust as appropriate.

Thanks,

Denis.

**Comment by GuopingJia** [comment]
Thanks Denis !

The new patch has addressed your comments on the Conclusion section. 

Guoping

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+* provide message queue performance and consuming latency monitoring;
+* start tracing from browser for browser performance monitoring;
+* support infrastructure monitoring for Kubernetes and Linux;
+* support alerting using rules in both observability analysis language and metric analysis language;
+
+Apache SkyWalking provides a list of agents to be used for building *Java*, *.NET Core*, *PHP*, *Node.js*, *Golang*, *LUA*, *Rust* and *C++* apps. It supports to integrate and collect data from multiple sources, including _Prometheus_, _OpenTelemetry_ and _Zabbix_ for metrics and logs, _Zipkin_ for traces. It provides tracing, metrics analysis, alerting, service mesh observability and visualization. 
+
+Apache SkyWalking is an open source tool with *21K* GitHub stars and *6K* GitHub forks. In [StackShare community](https://stackshare.io/apache-skywalking#stacks.), Apache SkyWalking does not yet have many approval, only being  mentioned in 12 developer stacks. However, Apache SkyWalking has more than *600* contributors on GitHub and thousands of contributions every year. All the agents for application instrumentation have been actively maintained.
```
Comment (drensing) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:214]: In the first sentence, change "does not yet have many approval" to something like "does not yet have much approval" or "does not yet have as broad an acceptence, only being..." or "does not yet have as great a share of the market..." (that last one might be the most appropriate.

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+Apache SkyWalking is the first open source project that initialized and implemented an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of _RPC_, this observation solution is much easier to be added without language specific technology. With this solution, users could get the application service topology map, metrics graph, request details and error message with a very nice visualization. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across on-premises, cloud-based or hybrid environments.
```
Comment (drensing) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:216]: 2nd sentence, add a hyphen between "language" and "specific" so it says "... language-specific technology."

3rd sentence, change "users could get" to "users **can** get".

Last sentence, change to "for monitoring and visualizing applicatio**ns** that **consist** of..."

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+Apache SkyWalking is the first open source project that initialized and implemented an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of _RPC_, this observation solution is much easier to be added without language specific technology. With this solution, users could get the application service topology map, metrics graph, request details and error message with a very nice visualization. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across on-premises, cloud-based or hybrid environments. 
+
+Apache SkyWalking is lightweight, scalable, and supports alerting and visualization. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and removes the restricted security restriction in user environment.
```
Comment (drensing) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:218]: Rewrite the first sentence to "Apache SkyWalking is lightweight and scalable, and it supports alerting and visualization." Or, divide it out into two sentences.

Last sentence, add a hyphen between "third" and "party". Change the ending to "...in **the user's** environment."

File: content/blog/get-started-with-application-performance-monitoring-tools-overview.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+Apache SkyWalking is lightweight, scalable, and supports alerting and visualization. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and removes the restricted security restriction in user environment. 
+
+## Conclusion
+
+This blog post gave a closer look at some of the best APM tools, either open-source or commercial vendor based. It listed the key features of each APM tool, and discussed in details their strengths and weaknesses. The importance of a good APM solution is now indisputable. All it takes is to pick the right one based on your monitoring needs for your applications.
```
Comment (drensing) [inline content/blog/get-started-with-application-performance-monitoring-tools-overview.md:222]: Suggested rephrasing for the conclusion:
"In this blog post, I hope I gave you a closer look at some of the best APM tools that are out there, both open-source and commercial vendor based. In it, I have listed the key features of each APM tool and discussed each toll's strengths and weaknesses." (you can keep the next two sentences just as they are :-) )

**Review by drensing — COMMENTED** [review]
I did have a lot of comments in this blog post, but I hope they help you. This is such a great , informative post - I just want it to be crystal clear in everything you are trying to say. (and, to be fair, it was a long blog post anyway :-) 

All the best,
Dale

**Comment by GuopingJia** [comment]
> I did have a lot of comments in this blog post, but I hope they help you. This is such a great , informative post - I just want it to be crystal clear in everything you are trying to say. (and, to be fair, it was a long blog post anyway :-)
> 
> All the best, Dale

Hi, Dale, 

Thank you for spending time reviewing my blog. Yes, it's a very long blog post and I spent a lot of time walking through all related tools, understanding/comparing the difference. Huge amount of contents could easily get me lost. When I was finished it, I was thinking about your review comments. I appreciate your comments and a long list of very good suggestions !

The new patch should address all your comments. But let me know if there is anything you want me to change.

Guoping

**Review by drensing — COMMENTED** [review]
Thank you for incorporating my requested changes! I know there were a lot, but you did a great job.

All the best,
Dale

**Comment by GuopingJia** [comment]
The new patch has addressed one comment from Fred.

**Pull request merged**